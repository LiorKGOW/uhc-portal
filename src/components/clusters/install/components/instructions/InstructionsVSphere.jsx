import React from 'react';
import PropTypes from 'prop-types';
import { Card, Title } from '@patternfly/react-core';
import links from '../../../../../common/installLinks';
import GetStarted from './components/GetStarted';
import TelemetryAlert from './components/TelemetryAlert';
import TokenErrorAlert from './components/TokenErrorAlert';
import DownloadsAndPullSecretSection from './components/DownloadsAndPullSecretSection';
import EvaluationAlert from './components/EvaluationAlert';

function InstructionsVSphere({ token }) {
  return (
    <>
      <Title headingLevel="h3" size="2xl">
        Install on vSphere with user-provisioned infrastructure
      </Title>
      <Card>
        <div className="pf-l-grid pf-m-gutter ocm-page">
          {token.error && <TokenErrorAlert token={token} />}
          <EvaluationAlert />
          <div className="pf-c-content">
            <p>
              With OpenShift Container Platform you can install a cluster on vSphere using
              infrastructure that you provide.
            </p>

            <GetStarted docURL={links.INSTALL_VSPHERE_GETTING_STARTED} />
          </div>
          <TelemetryAlert />
        </div>
      </Card>
      <DownloadsAndPullSecretSection
        token={token}
        installerURL={links.INSTALL_VSPHERE_INSTALLER_LATEST}
        rhcosLearnMoreURL={links.INSTALL_VSPHERE_RHCOS_LEARN_MORE}
        cliURL={links.INSTALL_VSPHERE_CLI_LATEST}
      />
    </>
  );
}

InstructionsVSphere.propTypes = {
  token: PropTypes.object.isRequired,
};

export default InstructionsVSphere;
