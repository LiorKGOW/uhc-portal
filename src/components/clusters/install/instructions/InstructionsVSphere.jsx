import React from 'react';
import PropTypes from 'prop-types';
import { Card, Title } from '@patternfly/react-core';
import GetStarted from './components/GetStarted';
import TelemetryAlert from './components/TelemetryAlert';
import TokenErrorAlert from './components/TokenErrorAlert';
import DownloadsAndPullSecretSection from './components/DownloadsAndPullSecretSection';
import EvaluationAlert from './components/EvaluationAlert';
import links, { channels } from '../../../../common/installLinks';

function InstructionsVSphere({ token }) {
  const cloudProviderID = window.location.pathname;
  return (
    <>
      <Title headingLevel="h3" size="2xl">
        Install on vSphere with user-provisioned infrastructure
      </Title>
      <Card>
        <div className="pf-l-grid pf-m-gutter ocm-page">
          {token.error && <TokenErrorAlert token={token} />}
          <EvaluationAlert />
          <div className="pf-c-content">
            <p>
              With OpenShift Container Platform you can install a cluster on vSphere using
              infrastructure that you provide.
            </p>

            <GetStarted
              docURL={links.INSTALL_VSPHERE_GETTING_STARTED}
              cloudProviderID={cloudProviderID}
            />
          </div>
          <TelemetryAlert />
        </div>
      </Card>
      <DownloadsAndPullSecretSection
        token={token}
        rhcosLearnMoreURL={links.INSTALL_VSPHERE_RHCOS_LEARN_MORE}
        cloudProviderID={cloudProviderID}
        channel={channels.STABLE}
      />
    </>
  );
}

InstructionsVSphere.propTypes = {
  token: PropTypes.object.isRequired,
};

export default InstructionsVSphere;
