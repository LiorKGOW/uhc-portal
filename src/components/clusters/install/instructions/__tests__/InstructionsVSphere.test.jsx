import React from 'react';
import { shallow } from 'enzyme';

import InstructionsVSphere from '../InstructionsVSphere';

describe('VSphere instructions', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InstructionsVSphere />);
    expect(wrapper).toMatchSnapshot();
  });
});
