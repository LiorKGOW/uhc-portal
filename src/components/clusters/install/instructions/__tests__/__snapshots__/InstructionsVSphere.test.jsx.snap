// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSphere instructions renders correctly 1`] = `
<Card>
  <div
    className="pf-c-content ocm-page"
  >
    <Title
      headingLevel="h3"
      size="2xl"
    >
      VMware vSphere: Select an installation type
    </Title>
    <div
      className="flex-container"
    >
      <Link
        className="ocm-c-ipi-upi-infra-card infra-card pf-c-card"
        data-testid="assisted-installer-vsphere-card"
        to="/assisted-installer/clusters/~new"
      >
        <CardBadge
          devPreviewText="Technology Preview"
          isDevPreview={true}
        />
        <CardBody
          className="ocm-c-ipi-upi-infra-card--body"
        >
          <ConnectedIcon
            alt="Installer-Provisioned Infrastructure"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
          <Title
            headingLevel="h3"
            size="lg"
          >
            Assisted Installer
          </Title>
           
          Install OpenShift on your own infrastructure with step-by-step guidance.
        </CardBody>
      </Link>
      <Link
        className="ocm-c-ipi-upi-infra-card infra-card pf-c-card"
        to="/install/vsphere/installer-provisioned"
      >
        <CardBadge
          isRecommened={true}
        />
        <CardBody
          className="ocm-c-ipi-upi-infra-card--body"
        >
          <SyncAltIcon
            alt="Installer-Provisioned Infrastructure"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
          <Title
            headingLevel="h3"
            size="lg"
          >
            Installer-provisioned infrastructure
          </Title>
           
          Deploy an OpenShift cluster on infrastructure that the installation program provisions and the cluster maintains.
        </CardBody>
      </Link>
      <Link
        className="ocm-c-ipi-upi-infra-card infra-card pf-c-card"
        to="/install/vsphere/user-provisioned"
      >
        <CardBadge
          isHidden={true}
        />
        <CardBody
          className="ocm-c-ipi-upi-infra-card--body"
        >
          <UserIcon
            alt="User-Provisioned Infrastructure"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
          <Title
            headingLevel="h3"
            size="lg"
          >
            User-provisioned infrastructure
          </Title>
          Deploy an OpenShift cluster on infrastructure that you prepare and maintain.
        </CardBody>
      </Link>
    </div>
  </div>
</Card>
`;
